### ISSUE TITLE
[ProvisioningStateMustBeReadOnly] Rule ignores siblings of $ref

### ISSUE THREAD

**Issue by cataggar** [Azure/azure-openapi-validator#637]
I am working on https://github.com/Azure/azure-rest-api-specs-pr/pull/15631 and there are many instances of `ProvisioningStateMustBeReadOnly` errors that are false positives.
https://github.com/Azure/azure-rest-api-specs-pr/pull/15631/checks?check_run_id=19650955197

All of the `provisioningState`s are marked as `readOnly`, for example:
```
        "provisioningState": {
          "$ref": "#/definitions/AddonProvisioningState",
          "description": "The state of the addon provisioning",
          "readOnly": true
        }
```

I see this test was considered flaky before in #561. cc @AkhilaIlla

**Comment by cataggar** [comment]
I saw this discussed in https://github.com/Azure/azure-sdk-tools/issues/6191. The solution was to add `use-read-only-status-schema: true` to the `tspconfig.yml`:

``` yml
options:
  "@azure-tools/typespec-autorest":
    azure-resource-provider-folder: "resource-manager"
    emitter-output-dir: "{project-root}/.."
    examples-directory: examples
    output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/vmware.json"
    omit-unreachable-types: true
    use-read-only-status-schema: true
```

**Comment by mikeharder** [comment]
`ProvisioningStateMustBeReadOnly` is a "spectral" rule:

https://github.com/Azure/azure-openapi-validator/blob/d37228967e2d5333a2a6196702d78ce3e4f1e222/packages/rulesets/src/spectral/az-arm.ts#L208-L221

It's probably ignoring the `readonly` property, because the OpenAPI 2.0 spec says it should be ignored (https://github.com/Azure/azure-openapi-validator/discussions/706).

It **might** be possible to configure this rule to allow siblings of $ref.
